protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -387,9 +387,9 @@ class _FieldSet {
   }
 
   /// The implementation of a generated getter for map fields.
-  Map<K, V> _$getMap<K, V>(GeneratedMessage parentMessage, int index) {
+  PbMap<K, V> _$getMap<K, V>(GeneratedMessage parentMessage, int index) {
     final value = _values[index];
-    if (value != null) return value as Map<K, V>;
+    if (value != null) return value as PbMap<K, V>;
```

Comment:
nit: For consistency: Either manually downcast with `as` (as here) or rely on auto-inserted downcasts (like in `_$getList<T>` above). The reason we support implicit casts is because the `_values` is `List<dynamic` (if it was `List<Object?>` then one would be forced to downcast explicitly all uses)

Reply:
Good catch. Updated this with an implicit cast.